weblm/controller.py

```diff
@@ -541,45 +380,24 @@ def pick_action(self, url: str, page_elements: List[str], response: str = None):
                               "Please respond with 'y' or 'n'")
             elif re.match(r'search (.+)', response):
                 query = re.match(r'search (.+)', response).group(1)
-                results = search(self.co, query, self._page_elements, topk=50)
+                results = self.cohere_controller.search( query, self._page_elements, topk=50)
                 return Prompt(f"Query: {query}\nResults:\n{results}\n\n"
                               "Please respond with 'y' or 'n'")
             else:
                 return Prompt("Please respond with 'y' or 'n'")
 
             self._step = DialogueState.Command
 
-    def _get_cmd_prediction(self, prompt: str, chosen_element: str) -> str:
-        if "type" in self._action:
-            text = None
-            while text is None:
-                try:
-                    num_tokens = 20
-                    if len(self.co.tokenize(prompt)) > 2048 - num_tokens:
-                        print(f"WARNING: truncating sequence of length {len(self.co.tokenize(prompt))}")
-                        prompt = truncate_left(self.co.tokenize,
-                                               prompt,
-                                               self._action,
-                                               chosen_element,
-                                               limit=2048 - num_tokens)
-
-                    print(len(self.co.tokenize(prompt + self._action + chosen_element)))
-                    text = max(self.co.generate(prompt=prompt + self._action + chosen_element,
-                                                model=MODEL,
-                                                temperature=0.5,
-                                                num_generations=5,
-                                                max_tokens=num_tokens,
-                                                stop_sequences=["\n"],
-                                                return_likelihoods="GENERATION").generations,
-                               key=lambda x: x.likelihood).text
-                    print(text)
-                except cohere.error.CohereError as e:
-                    print(f"Cohere fucked up: {e}")
-                    continue
-        else:
-            text = ""
+    
 
-        return (self._action + chosen_element + text).strip()
+    # this comment explains the function below
```

Comment:
Move this inside the function
```suggestion
    def generate_command(self, url: str, pruned_elements: List[str], response: str = None):
        """The goal is to generate a command that is as short as possible, but still
        contains all the information necessary to execute the command
        the strategy is to generate a command, then check if it is valid
        if it is valid, then we're done
        if it is not valid, then we try to remove the last element of the command
        and check if the resulting command is valid
        if it is valid, then we're done"""
        state = self._construct_state(url, pruned_elements)
```